netsetup/wpantun.py

```diff
@@ -37,20 +37,41 @@ class WpanClient(object):
     __metaclass__ = Singleton
 
     bus = None
-    iface = None
-    iface_state = None
-
-    state = ""
-    node_type = ""
-    network_name = ""
-    pan_id = 0
-    channel = 0
-    xpan_id = ""
-    mesh_ipv6 = ""
-    masterkey = ""
 
     def __init__(self):
         self.bus = dbus.SystemBus()
+        self.iface = None
+        self.iface_state = None
+        self.signal_match = None
+        self.state = ""
+        self.node_type = ""
+        self.network_name = ""
+        self.pan_id = 0
+        self.channel = 0
+        self.xpan_id = ""
+        self.mesh_ipv6 = ""
+        self.masterkey = ""
+
+    def _on_name_owner_changed(self, newOwner):
+        if not newOwner:
+            logging.info("wpantund is down")
+            self._stop_monitor_properties_changes()
+        else:
+            logging.info("wpantund is up")
+            self._start_monitor_properties_changes()
+
+    def start(self):
```

Comment:
You can put the start() before _on_name_owner_changed or
_start_monitor_properties_changes and _stop_monitor_properties_changes before _on_name_owner_changed.

Reply:
Done